libscidavis/src/Convolution.cpp

```diff
@@ -93,32 +88,31 @@ void Convolution::setDataFromTable(Table *t, const QString &signalColName,
         return;
     }
 
-    d_n = rows;
-
     d_n_signal = 16; // tmp number of points
-    while (unsigned(d_n_signal) < d_n + d_n_response / 2)
+    while (unsigned(d_n_signal) < rows + d_n_response / 2)
         d_n_signal *= 2;
 
-    d_x = new double[d_n_signal]; // signal
-    d_y = new double[d_n_response]; // response
-
-    if (d_y && d_x) {
-        memset(d_x, 0, d_n_signal * sizeof(double)); // zero-pad signal data array
-        for (unsigned i = 0; i < d_n; i++)
-            d_x[i] = d_table->cell(i, signal_col);
-        for (int i = 0; i < d_n_response; i++)
-            d_y[i] = d_table->cell(i, response_col);
-    } else {
+    try {
+        d_x.resize(d_n_signal); // signal
+        d_y.resize(d_n_response); // response
+    } catch (const std::bad_alloc &e) {
         QMessageBox::critical((ApplicationWindow *)parent(), tr("SciDAVis") + " - " + tr("Error"),
-                              tr("Could not allocate memory, operation aborted!"));
+                              tr("Could not allocate memory, operation aborted!\n")
+                                      + tr("Allocator returned: ") + e.what());
         d_init_err = true;
-        d_n = 0;
+        return;
     }
+    std::fill(d_x.begin(), d_x.end(), 0.0);
+    std::fill(d_y.begin(), d_y.end(), 0.0);
+    for (unsigned i = 0; i < d_n_signal; i++)
```

Comment:
This loop bound should be rows, as d_n_signal may be too large for the following cell() call.